### DISCUSSION TITLE
Include relationship objects in responses?

### DISCUSSION THREAD

**Discussion by vikt0r0** [omar-dulaimi/prisma-trpc-generator#11] [Q&A] [answered]
Hi Omar, thanks for this amazing package. Is it somehow possible to fetch related objects in find queries?

Currently it does not seem that a schema is generated for such, or am I misunderstood?

If I understand, it would imply extending the (?)Find(Unique|Many)Schema?

`export const (?)FindUniqueSchema = z.object({ where: (?)WhereUniqueInputObjectSchema });`

Are you working on this currently, or is it a non-issue? If you accept pull requests, I may have a look if I have the time.

Best.

**Comment by omar-dulaimi** [comment]
Hello @vikt0r0 

Is this what you mean?

```
const getUser = await prisma.user.findUnique({
  where: {
    id: 19,
  },
  include: {
    posts: true,
  },
});
```

This basically means that `include` types are currently missing.

**Comment by vikt0r0** [comment]
Hi Omar,

Indeed - I am fairly new to TypeScript and the React environment, so
forgive me if I’m a bit slow on picking up.

But this means currently one would have to make several requests or write
my own routers, if I understand correctly?


Best.

fre. 15. jul. 2022 kl. 15.50 skrev Omar Dulaimi ***@***.***>:

**Reply by omar-dulaimi** [reply]
Not at all, we all start somewhere, nobody is born with all the knowledge.
Also, you are right. Currently you have to issue more requests to get the nested items.

I'm going to create an issue from this discussion. Hoping I or somebody else tackle this feature request.

**Comment by omar-dulaimi** [comment, marked as answer]
Feature request issue created here: https://github.com/omar-dulaimi/prisma-trpc-generator/issues/13

**Reply by vikt0r0** [reply]
Very cool; would you be interested in some assistance with the implementation? I currently have some extra time on my hand.

**Reply by omar-dulaimi** [reply]
Yes, of course. Though since I'm currently looking for a job, I might not be available much. Let me know how I can help.

**Comment by vikt0r0** [comment]
Cool. I’m actually also applying these days, hence I have a little more spare time on my hands than usually.

If you are up for it we could set up a call sometime and perhaps you could give me your take on how you think I should approach it?
